misk-service/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -14,6 +15,31 @@ internal class CoordinatedService(
 
   override val service: Service by lazy {
     serviceProvider.get()
+      .also { created ->
+        created.checkNew("$created must be NEW for it to be coordinated")
+        created.addListener(
+          object : Listener() {
+            override fun running() {
+              synchronized(this@CoordinatedService) {
```

Comment:
I added the `@CoordinatedService` bit, which I think was always intended but not included.

Reply:
yup, that makes sense to me! :)